executor/src/core/LanguageController.ts

```
@@ -601,34 +601,54 @@ export default class LanguageController {
             //TODO: here was are assuming that the first character is " and the last two "; we should check for this and not assume
-            dnaCode = dnaCode.substring(1, dnaCode.length-2);
-            fs.writeFileSync(tempDnaPath, Buffer.from(dnaCode, "base64"));
-
-            //Unpack the DNA
+            happCode = happCode.substring(1, happCode.length-2);
+            fs.writeFileSync(tempHappPath, Buffer.from(happCode, "base64"));
+            //fs.writeFileSync(tempHappPath+"-original", Buffer.from(happCode, "base64"));
+            //Unpack hApp bundle
             //TODO: we need to be able to check for errors in this fn call, currently we just crudly split the result
+            console.log("LanguageController.readAndTemplateHolochainDNA: unpacking hApp bundle");
+            let unpackHappPath = (await this.#holochainService?.unPackHapp(tempHappPath)).replace(/(\r\n|\n|\r)/gm, "");
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)